modules/ROOT/pages/patterns/fixed-length-patterns.adoc

```diff
@@ -0,0 +1,223 @@
+----
+MATCH (n)
+----
+
+With this reference, node properties can be accessed:
+
+[source, role=noheader]
+----
+MATCH (n)
+RETURN n.name
+----
+
+Adding a label expression to the node pattern means only nodes with labels that match will be returned.
+The following matches nodes that have the `Stop` label:
+
+[source, role=noheader]
+----
+MATCH (n:Stop)
+----
+
+The following more complex label expression matches all nodes that are either a `TrainStation` and a `BusStation` or `StationGroup`:
+
+[source, role=noheader]
+----
+MATCH (n:(TrainStation&BusStation)|StationGroup)
+----
+
+A map of property names and values can be used to match on node properties based on equality with the specified values.
+The following matches nodes that have their `mode` property equal to `Rail`:
+
+[source, role=noheader]
+----
+MATCH (n { mode: 'Rail' })
+----
+
+More general predicates can be expressed with a `WHERE` clause.
+The following matches nodes whose name property starts with `Preston`:
+
+[source, role=noheader]
+----
+MATCH (n:Station WHERE n.name STARTS WITH 'Preston')
+----
+
+See the xref:patterns/reference.adoc#node-patterns[node patterns] reference section for more details.
+
+[[relationship-patterns]]
+== Relationship patterns
+
+The simplest possible relationship pattern is a pair of dashes:
+
+[source, role=noheader]
+----
+--
+----
+
+This pattern matches a relationship with any direction and does not filter on any relationship type or property.
+Unlike a node pattern, a relationship pattern cannot be used in a `MATCH` clause without node patterns at both ends.
+See xref:patterns/fixed-length-patterns.adoc#path-patterns[path patterns] for more details.
+
+In order to obtain a reference to the relationships matched by the pattern, a relationship variable needs to be declared in the pattern by adding the variable name in square brackets in between the dashes:
+
+[source, role=noheader]
+----
+-[r]-
+----
+
+To match a specific direction, add `<` or `>` to the left or right hand side respectively:
+
+[source, role=noheader]
+----
+-[r]->
+----
+
+To match on a relationship type, add the type name after a colon:
+
+[source, role=noheader]
+----
+-[:CALLS_AT]->
+----
+
+Similar to node patterns, a map of property names and values can be added to filter on properties of the relationship based on equality with the specified values:
+
+[source, role=noheader]
+----
+-[{ distance: 0.24, duration: 'PT4M' }]->
+----
+
+A `WHERE` clause can be used for more general predicates:
+
+[source, role=noheader]
+----
+-[r WHERE time() + duration(r.duration) < time('22:00') ]->
+----
+
+See the xref:patterns/reference.adoc#relationship-patterns[relationship patterns] reference section for more details.
+
+[[path-patterns]]
+== Path patterns
+
+Any valid path starts and ends with a node, with relationships between each node (if there is more than one node).
+Path patterns have the same restrictions, and for all valid path patterns the following are true:
```

Comment:
If we have parenthesized path patterns with juxtaposition, that is no longer the case. Do we therefore paint ourselves into a corner by writing it like this?